src/lib/utils/ct_utils.h

```diff
@@ -265,11 +280,16 @@ constexpr inline T value_barrier(T x) {
       * (see https://theunixzoo.co.uk/blog/2021-10-14-preventing-optimisations.html)
       * however the current approach seems sufficient with current compilers,
       * and is minimally damaging with regards to degrading code generation.
+      *
+      * Note that the asm-based approach is disabled on SuperH because it caused
+      * miscompilation on GCC 13.2. See GH #4444 for more details.
       */
-#if defined(BOTAN_USE_GCC_INLINE_ASM) && !defined(BOTAN_HAS_SANITIZER_MEMORY)
+#if defined(BOTAN_USE_GCC_INLINE_ASM) && !defined(BOTAN_HAS_SANITIZER_MEMORY) && !defined(BOTAN_TARGET_ARCH_IS_SUPERH)
       asm("" : "+r"(x) : /* no input */);
-#endif
       return x;
+#else
+      return volatile_read(x);
```

Comment:
This introduces it for MSVC (and others) as well which maybe is the “right” thing to do but I could see this causing a massive performance regression and AFAIK MSVC is not known to ever perform Clang-style optimizations that undo constant time code.

Likewise it introduces the volatile reads for the sanitizer build which doesn’t seem necessary; in general MemSan isn’t safe for production builds anyway.

I think we need A Couple More Macros (tm) here

* `BOTAN_COMPILER_USE_CT_BARRIER` - possibly set from the compiler info file? For GCC, Clang, XCode, maybe others.
* `BOTAN_COMPILER_CT_BARRIER_USE_ASM_BLOCK`
* `BOTAN_COMPILER_CT_BARRIER_USE_VOLATILE`

Maybe even toggleable at configure time? (`none`, `asm`, `volatile`) OK at this point I’m convinced this is worthwhile but I can take it over the holiday weekend …

Reply:
For the record: #4447